plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeTableChangesWithAccessTracking.java

```diff
@@ -0,0 +1,96 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.Session;
+import io.trino.filesystem.hdfs.HdfsFileSystemFactory;
+import io.trino.plugin.deltalake.metastore.TestingDeltaLakeMetastoreModule;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Optional;
+
+import static com.google.inject.util.Modules.EMPTY_MODULE;
+import static io.trino.plugin.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static io.trino.plugin.hive.metastore.file.FileHiveMetastore.createTestingFileHiveMetastore;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Test(singleThreaded = true)
+public class TestDeltaLakeTableChangesWithAccessTracking
+        extends AbstractTestQueryFramework
+{
+    protected static final String SCHEMA = "test_table_changes_tracking_" + randomNameSuffix();
+
+    private static final Session TEST_SESSION = testSessionBuilder()
+            .setCatalog("delta_lake")
+            .setSchema(SCHEMA)
+            .build();
+
+    private File dataDirectory;
+    private HiveMetastore metastore;
+    private AccessTrackingFileSystemFactory fileSystemFactory;
+
+    @Override
+    protected DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(TEST_SESSION).build();
+
+        dataDirectory = queryRunner.getCoordinator().getBaseDataDir().resolve("delta_lake").toFile();
+        metastore = createTestingFileHiveMetastore(dataDirectory);
+        fileSystemFactory = new AccessTrackingFileSystemFactory(new HdfsFileSystemFactory(HDFS_ENVIRONMENT));
+
+        queryRunner.installPlugin(new TestingDeltaLakePlugin(Optional.of(new TestingDeltaLakeMetastoreModule(metastore)), Optional.of(fileSystemFactory), EMPTY_MODULE));
+        queryRunner.createCatalog(
+                "delta_lake",
+                "delta_lake",
+                ImmutableMap.of("delta.enable-non-concurrent-writes", "true"));
+
+        queryRunner.execute("CREATE SCHEMA " + SCHEMA);
+        return queryRunner;
+    }
+
+    @Test
+    public void testTableChangesFileSystemAccess()
+    {
+        String tableName = "table_changes_" + randomNameSuffix();
```

Comment:
we're operating in isolation, so the table name can be fixed & inlined.
this would improve readability for the test queries (no need for string concatenation)